File: src/dataset/python/protein-translation_exercism-fruit-in/gemini/GeminiFlash_protein_translation_v1.py
CODONS = {
    "AUG": "Methionine", "UUU": "Phenylalanine", "UUC": "Phenylalanine",
    "UUA": "Leucine", "UUG": "Leucine", "UCU": "Serine", "UCC": "Serine",
    "UCA": "Serine", "UCG": "Serine", "UAU": "Tyrosine", "UAC": "Tyrosine",
    "UGU": "Cysteine", "UGC": "Cysteine", "UGG": "Tryptophan",
    "UAA": "STOP", "UAG": "STOP", "UGA": "STOP",
}


def proteins(strand):
    proteins_list = []
    for i in range(0, len(strand), 3):
        codon = strand[i:i + 3]
        protein = CODONS.get(codon)
        if protein == "STOP":
            break
        if protein:
            proteins_list.append(protein)
    return proteins_list

File: src/dataset/javascript/protein-translation_exercism-ThomasZumsteg/gemini/GeminiFlash_protein-translation_v1.js
const translate = {
    'AUG': 'Methionine',
    'UCA': 'Serine',
    'UCC': 'Serine',
    'UCG': 'Serine',
    'UCU': 'Serine',
    'UUA': 'Leucine',
    'UUC': 'Phenylalanine',
    'UUG': 'Leucine',
    'UUU': 'Phenylalanine',
    'UAU': 'Tyrosine',
    'UAC': 'Tyrosine',
    'UGC': 'Cysteine',
    'UGU': 'Cysteine',
    'UGG': 'Tryptophan',
    'UAA': 'STOP',
    'UAG': 'STOP',
    'UGA': 'STOP',
};

module.exports = function(rna) {
    if (!rna) {
        return [];
    }

    const result = [];
    for (let i = 0; i < rna.length; i += 3) {
        const codon = rna.slice(i, i + 3);
        const protein = translate[codon];

        if (protein === 'STOP') {
            break;
        }

        if (!protein) {
            throw new Error('Invalid codon');
        }

        result.push(protein);
    }

    return result;
};